guvnor-webapp/src/main/java/org/guvnor/client/perspectives/M2RepoPerspective.java

```diff
@@ -20,13 +20,13 @@
 import javax.enterprise.event.Event;
```

Comment:
Heh.. now you don't destroy the bean you created when the "Upload" popup closes!

You need to do something like this:

```
UploadFormPresenter uploadFormPresenter = iocManager.lookupBean( UploadFormPresenter.class ).getInstance();
uploadFormPresenter.getView().addCloseHandler( new CloseHandler<PopupPanel>() {

                            @Override
                            public void onClose( CloseEvent<PopupPanel> event ) {
                                iocManager.destroyBean( uploadFormPresenter );
                            }

                        } );
```
Or something like that!